core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RangeValueFilterExecuterImpl.java

```diff
@@ -511,7 +511,7 @@ private BitSet setFilterdIndexToBitSetWithColumnIndex(
 
     // Binary Search cannot be done on '@NU#LL$!", so need to check and compare for null on
     // matching row.
-    if (dimensionColumnPage.isNoDicitionaryColumn()) {
+    if (dimensionColumnPage.isNoDicitionaryColumn() && !dimensionColumnPage.isAdaptiveEncoded()) {
```

Comment:
Not related to this PR. But I think we better have a function to return the encoding type of the columnPage instead of having `isAdaptiveEncoded`, since we will add more encoding in the future. 
@ravipesala @ajantha-bhat  please check

Reply:
@jackylk isAdaptiveEncoded() is already set/decided by the function org.apache.carbondata.core.datastore.chunk.reader.dimension.v3.CompressedDimensionChunkFileBasedReaderV3#isEncodedWithAdaptiveMeta. In future also we can change the function accordingly

Reply:
@jackylk : yes, we can have a BitSet in columnPage for enum of Encoding, and from PageMetaData encodings we can set the bitset. And to check whether particular encoding is there or not, we just have to check whether that bit is set or not.

Currently we don't need this, but like you said, it will help for future changes. I can do this change if you and @ravipesala  agrees for this change.